inline-file.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+sed -e "/$1/ {
+r $3
+d }" <"$2"
```

Comment:
Any reason why this needs to be a script and can't be inlined in the Makefile?

Reply:
That was my initial inclination, but didn't manage to make it work. If you have anything that works I'll be happy to respin/integrate.

Reply:
Ack, removed the m4 -> sed change for now.

For some context, the switch was based on a number of factors. Mentioning some for posterity:
 - declining adoption and knowledge of m4
 - single implementation - gnu m4 vs bsd/gnu sed
 - very infrequent releases - v1.4.19 (2 years ago), .18 (6 years) .17 (10 years)
 - seriously stagnated development - 1.4 branch, ~50 meaningful changes over 10 years, 1.6 branch AWOL for 13 years, 2.0 branch AWOL for 6 years

Reply:
Yep, I agree. Maybe we should opt for an `arch-install-scripts` library thing like devtools/mkinitcpio/makepkg does instead.

However that can be a seperate PR i think.

Reply:
The library approach makes sense when there is notable amount of shared code (and thus duplication) or a test suite doing unit checking individual functionality.

Currently we're at 100 loc and zero tests - so quite far off on both fronts.

Just pushed an update after re-testing everything - genfstab was off :facepalm:

Reply:
That's not true, we do have tests.

https://github.com/archlinux/arch-install-scripts/tree/master/test

I don't see why that would be an arbitrary bar to cross though :)

Reply:
I feel like we're going off a tangent here?